src/layout/msWorkspace/msWindow.ts

```diff
@@ -940,8 +1134,10 @@ export class MsWindow extends Clutter.Actor {
     }
 
     _onDestroy() {
-        this.destroyed = true;
+        if (this.destroyId) this.disconnect(this.destroyId);
+        this.msWorkspace.removeMsWindow(this);
```

Comment:
Is wrong in both case but they are not equivalent either.
the kill method is an action done by the user where _onDestroy is called on any cases maybe could cause some bugs

Reply:
It seems odd to me that we shouldn't remove the ms window from the workspace, just in case the window was killed in another way. How would it be removed then?

Reply:
I'm not sure but I remember of a bug when shutting down the computer or logout where all windows was closing and trying to be removed from a msWorkspace already gone and crashed up the state persistency